exchanges/kraken/kraken_wrapper.go

```diff
@@ -463,49 +549,71 @@ func (k *Kraken) UpdateOrderbook(p currency.Pair, assetType asset.Item) (*orderb
 
 // UpdateAccountInfo retrieves balances for all enabled currencies for the
 // Kraken exchange - to-do
-func (k *Kraken) UpdateAccountInfo() (account.Holdings, error) {
+func (k *Kraken) UpdateAccountInfo(assetType asset.Item) (account.Holdings, error) {
 	var info account.Holdings
-	info.Exchange = k.Name
-
-	bal, err := k.GetBalance()
-	if err != nil {
-		return info, err
-	}
-
 	var balances []account.Balance
-	for key := range bal {
-		translatedCurrency := assetTranslator.LookupAltname(key)
-		if translatedCurrency == "" {
-			log.Warnf(log.ExchangeSys, "%s unable to translate currency: %s\n",
-				k.Name,
-				key)
-			continue
+	info.Exchange = k.Name
+	switch assetType {
+	case asset.Spot:
+		bal, err := k.GetBalance()
+		if err != nil {
+			return info, err
 		}
-		balances = append(balances, account.Balance{
-			CurrencyName: currency.NewCode(translatedCurrency),
-			TotalValue:   bal[key],
+		for key := range bal {
+			translatedCurrency := assetTranslator.LookupAltname(key)
+			if translatedCurrency == "" {
+				log.Warnf(log.ExchangeSys, "%s unable to translate currency: %s\n",
+					k.Name,
+					key)
+				continue
+			}
+			balances = append(balances, account.Balance{
+				CurrencyName: currency.NewCode(translatedCurrency),
+				TotalValue:   bal[key],
+			})
+		}
+		info.Accounts = append(info.Accounts, account.SubAccount{
+			Currencies: balances,
 		})
+	case asset.Futures:
+		bal, err := k.GetFuturesAccountData()
+		if err != nil {
+			return info, err
+		}
+		for name := range bal.Accounts {
+			for code := range bal.Accounts[name].Balances {
+				translatedCurrency := assetTranslator.LookupAltname(strings.ToUpper(code))
+				if translatedCurrency == "" {
```

Comment:
Since you are not seeding the futures assets pairs in `FetchTradablePairs`, there will never be a translated currency. Check the test `TestFetchTradablePairs` with futures and verbose to see what I mean